vllm/model_executor/layers/fused_moe/layer.py

```diff
@@ -204,6 +224,9 @@ def __init__(
         self.num_expert_group = num_expert_group
         self.topk_group = topk_group
         self.custom_routing_function = custom_routing_function
+        if current_platform.is_hpu():
+            from vllm_hpu_extension.ops import StaticFusedMOE
+            self.hpu_static_fused_moe = StaticFusedMOE(self.num_experts)
```

Comment:
Could you please elaborate on how this works?

Reply:
This is a MoE implementation that contains only static shapes at the cost of tensors padding. When tokens are allocated to experts, tensors shapes are not determined by the specific allocation, instead they are aligned to token count. This was proposed to avoid the need to handle dynamic shapes for hpu implementation.

Reply:
We agreed to drop StaticFusedMOE from this PR's scope.